ios/FluentUI/Core/Theme/FluentTheme.swift

```diff
@@ -36,33 +36,24 @@ public class FluentTheme: NSObject, ObservableObject {
                 shadowOverrides: [ShadowToken: ShadowInfo]? = nil,
                 typographyOverrides: [TypographyToken: UIFont]? = nil,
                 gradientOverrides: [GradientToken: [UIColor]]? = nil) {
-        let fixedColorOverrides = colorOverrides?.map({ (key: ColorToken, value: UIColor) in
-            let newKey = AliasTokens.ColorsTokens(rawValue: key.rawValue)!
-            let newValue = value.dynamicColor!
-            return (newKey, newValue)
-        }) ?? [(AliasTokens.ColorsTokens, DynamicColor)]()
-
-        let fixedShadowOverrides = shadowOverrides?.map({ (key: ShadowToken, value: ShadowInfo) in
-            let newKey = AliasTokens.ShadowTokens(rawValue: key.rawValue)!
-            return (newKey, value)
-        }) ?? [(AliasTokens.ShadowTokens, ShadowInfo)]()
-
-        let fixedTypographyOverrides = typographyOverrides?.map({ (key: TypographyToken, value: UIFont) in
-            let newKey = AliasTokens.TypographyTokens(rawValue: key.rawValue)!
-            let newValue = FontInfo(name: value.fontName, size: value.pointSize)
-            return (newKey, newValue)
-        }) ?? [(AliasTokens.TypographyTokens, FontInfo)]()
-
-        let fixedGradientOverrides = gradientOverrides?.map({ (key: GradientToken, value: [UIColor]) in
-            let newKey = AliasTokens.GradientTokens(rawValue: key.rawValue)!
-            return (newKey, value)
-        }) ?? [(AliasTokens.GradientTokens, [UIColor])]()
-
-        // Pass overrides to AliasTokens
-        aliasTokens = .init(colorOverrides: Dictionary(uniqueKeysWithValues: fixedColorOverrides),
-                            shadowOverrides: Dictionary(uniqueKeysWithValues: fixedShadowOverrides),
-                            typographyOverrides: Dictionary(uniqueKeysWithValues: fixedTypographyOverrides),
-                            gradientOverrides: Dictionary(uniqueKeysWithValues: fixedGradientOverrides))
+
+        // Need to massage UIFonts into FontInfo objects
+        let mappedTypographyOverrides = typographyOverrides?.compactMapValues({ font in
+            return FontInfo(name: font.fontName, size: font.pointSize)
```

Comment:
Are we not able to just use UIFont for the typography overrides?

Reply:
I thought one of the issues we had with `FontInfo` was there wasn't a good way to guarantee the correct conversion of `UIFont` -> `FontInfo` -> `UIFont`, so seems like we shouldn't be going through `FontInfo` here and `defaultTypography` should return `UIFont`. Keeping `FontInfo` around for our internal inits makes sense, but should we also deprecate the public `FontInfo`? Likewise with `ColorValue` and `DynamicColor` if we're trying to move to just `UIColor`.

Reply:
So I do want to deprecate `ColorValue` and `DynamicColor`, and I even have a separate commit ready for that. It's a little wide ranging, so I wanted to keep it as a follow-up instead of the same PR, but if you prefer then I can fold it in here.

As for `FontInfo`, my intent was to avoid creating UIFont objects until they're needed -- my understanding previously was that they can be a bit heavy to instantiate, but that info might be a little outdated now. Do you have any recollection/feeling there @huwilkes?

Reply:
Separate PR for colors sounds good, as long as we can also get it in for this version bump.

For `FontInfo`, I don't recall that coming up in the previous `UIFont` vs `FontInfo` conversations, but I could see it being the case. As long as we can guarantee that we return the correct fonts that the clients pass in for their theme overrides I don't see an issue with continuing to use `FontInfo` here. We could also do some profiling tests to see how much instantiating `UIFont`s will cost us here.

Reply:
Discussed offline -- we're gonna dig deeper into the Font question at a later date, but as far as we can tell this doesn't regress our current situation, so let's move forward with it.